autotraining.lua

```diff
@@ -0,0 +1,290 @@
+-- Based on the original code by RNGStrategist (who also got some help from Uncle Danny)
+--@ enable = true
+--@ module = true
+
+local repeatUtil = require('repeat-util')
+local utils=require('utils')
+
+validArgs = utils.invert({
+    't'
+})
+
+local args = utils.processArgs({...}, validArgs)
+local GLOBAL_KEY  = "autotraining"
+local need_id = df.need_type['MartialTraining']
```

Comment:
I would call this constant `MartialTraining`, the current name seems awfully generic